var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -50,12 +63,81 @@ class Mesa(AutotoolsPackage):
     depends_on('presentproto@1.0:', type='build')
     depends_on('pkg-config@0.9.0:', type='build')
+    depends_on('binutils', type='build')
 
     # TODO: Add package for systemd, provides libudev
     # Using the system package manager to install systemd didn't work for me
+
+    def url_for_version(self, version):
+        """Handle mesa version-based custom URLs."""
+        latest_mver = '11'
+
+        if version < Version(latest_mver):
+            base = Mesa.base_url + "/older-versions/%s.x" % (version.up_to(1))
+        else:
+            base = Mesa.base_url 
+        if version > Version('10.4'):
+            name = 'mesa'
+        else:
+            name = 'MesaLib'
+        ret = base + "/%s/%s-%s.tar.gz" % (version, name, version)
+        return ret
+
+    def check_variants(self, var='', ver='', ref=''):
+        if var and ver:
+            error = "to be safe avoid '{variant}' with version '{version}' see:\n {reference}" # noqa
+            if var in self.spec and self.spec.satisfies(ver):
+                raise RuntimeError(
+                    error.format(variant=var, version=ver, reference=ref))
+
+#    def install(self, spec, prefix):
+#        configure('--prefix={0}'.format(prefix))
+#        self.check_variants(
+#            var='+gallium',
+#            ver='@:9.2.2',
+#            ref='http://www.paraview.org/Wiki/ParaView/ParaView_And_Mesa_3D')
+
+    def configure_args(self):
+        self.check_variants(
+            var='+gallium',
+            ver='@:9.2.2',
+            ref='http://www.paraview.org/Wiki/ParaView/ParaView_And_Mesa_3D')
+        args = []
+        if '+gallium' in self.spec:
+            args = ['--enable-opengl', '--disable-gles1', '--disable-gles2',
+                    '--disable-va', '--disable-xvmc', '--disable-vdpa', 
+                    '--enable-gallium-llvm',
+                    '--with-gallium-drivers=swrast,swr',
+                    '--disable-dri', '--with-dri-drivers=',
+                    '--disable-egl', '--with-egl-platforms=', '--disable-gb',
+                    '--disable-osmesa', '--enable-gallium-osmesa']
+            if '+shared_libs' in self.spec['llvm']:
+                args.append('--enable-llvm-shared-libs')
+            else:
+                args.append('--disable-llvm-shared-libs')
+            args.append('--with-llvm-prefix=' + self.spec['llvm'].prefix)
+        return args
+
+    def setup_environment(self, spack_env, run_env):
+        if '+gallium' in self.spec:
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'lib'))
+            run_env.set('GALLIUM_DRIVER', 'swr')
+
+    def setup_dependent_environment(self, spack_env, run_env, dspec):
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'lib'))
+            run_env.set('GALLIUM_DRIVER', 'swr')
+
+      
+#        options = ['-prefix=%s' % prefix]
```

Comment:
If I remember well, this is needed to activate the gallium driver that  is much faster than the others